### PR TITLE
Boos Details UI components implementation

### PR DESCRIPTION
**Pull request by pantelisss**
## **Why?**
Boost details screen
### **How?**
- Handle unknown boosts in the reward details screen
- Boost details screen implementation
- Integrated `nuke` library for async image load with caching
### **Testing**
Make sure boosts are presented properly using the dev server
### **Additional context**
fixes fe-663

### CODE REVIEW

**Review by PavlosTze — APPROVED** [review]
Good job @pantelisss . Everything looks OK. Found only one minor issue, probably an API issue as the response is random, but had to report it:

image

File: PresentationLayer/Extensions/Domain Extensions/NetworkDeviceRewardDetailsResponse+.swift [hunk 1]
```diff
@@ -215,3 +215,35 @@ extension NetworkDeviceRewardDetailsResponse {
...
+			  actualReward: 0.2442141,
+			  rewardScore: 71,
+			  maxReward: 1.0)
+	}
+
+	func toBoostViewObject(with date: Date?) -> BoostCardView.Boost {
+		.init(title: title ?? "",
+			  reward: actualReward ?? 0.0,
+			  date: date,
+			  score: rewardScore ?? 0,
```
Comment (PavlosTze) [inline PresentationLayer/Extensions/Domain Extensions/NetworkDeviceRewardDetailsResponse+.swift:235]: Hmm this looks wrong, as we want the `score` to be nullable in order to remove the texts on boost details screen if reward score is missing. And if I get it correctly that's not possible with this line here as it defaults to 0.

File: PresentationLayer/UI Components/Screens/Daily Rewards /Components/BoostsView.swift [hunk 2]
```diff
@@ -36,11 +37,12 @@ struct BoostsView: View {
...
-				image
-					.resizable()
-			} placeholder: {
-				ProgressView()
+			LazyImage(url: URL(string: imageUrl)) { state in
+				if let image = state.image {
+					image.resizable()
+				} else {
+					ProgressView()
+				}
```
Comment (PavlosTze) [inline PresentationLayer/UI Components/Screens/Daily Rewards /Components/BoostsView.swift:45]: I removed the `img_url` from the mock json and it crashed here. We need to have a fallback if `img_url` is missing, to add a custom hardcoded background, I've used `dark_bg` color for that.

**Review by PavlosTze — CHANGES_REQUESTED** [review]
Sorry @pantelisss , caught some issues in the other PR but I think this is one is more relevant to report them.

**Comment by pantelisss** [comment]
Great catches @PavlosTze

**Review by PavlosTze — APPROVED** [review]

**Pull request merged**